common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java

```diff
@@ -117,6 +118,8 @@
 
     private final HttpClient httpClient = UrlConnectionHttpClient.getDefaultInstance();
 
+    private AbstractAuthenticationScheme mAuthScheme;
```

Comment:
If this cannot be `final` then it should probably be removed, otherwise it can cause state issues.